pkg/helper/client.go

```diff
@@ -214,33 +214,37 @@ func checkApiServerConnectivity(timeout time.Duration) error {
 		config.Timeout = timeout
 		client, err := client.New(config, client.Options{})
 		if err != nil {
-			return false, errors.Wrap(err, "creating new custom client")
+			log.Error(err, "failed to creating new custom client")
+			return false, nil
 		}
 		err = client.Get(context.TODO(), types.NamespacedName{Name: metav1.NamespaceDefault}, &corev1.Namespace{})
 		if err != nil {
-			return false, errors.Wrap(err, "reaching to apiserver failed")
+			log.Error(err, "failed reaching the apiserver")
+			return false, nil
 		}
 		return true, nil
 	})
 }
 
 func defaultGw() (string, error) {
 	defaultGw := ""
-	return defaultGw, wait.PollImmediate(1*time.Second, defaultGwRetrieveTimeout, func() (bool, error) {
+	return defaultGw, wait.PollImmediate(1*time.Second, defaultGwRetrieveTimeout*time.Second, func() (bool, error) {
 		observedStateRaw, err := show()
 		if err != nil {
-			return false, fmt.Errorf("error running nmstatectl show: %v", err)
+			log.Error(err, fmt.Sprintf("failed retrieving current state"))
+			return false, nil
 		}
 
 		currentState, err := yaml.YAMLToJSON([]byte(observedStateRaw))
 		if err != nil {
-			return false, fmt.Errorf("Impossible to convert current state to JSON: %v", err)
+			return false, errors.Wrap(err, "failed to convert current state to JSON")
 		}
 
-		defaultGw = gjson.ParseBytes([]byte(currentState)).
+		defaultGw = gjson.ParseBytes(currentState).
 			Get("routes.running.#(destination==\"0.0.0.0/0\").next-hop-address").String()
 		if defaultGw == "" {
-			return false, fmt.Errorf("Impossible to retrieve default gw, state: %s", string(observedStateRaw))
+			log.Info("default gw missing", map[string]string{"state": string(currentState)})
```

Comment:
why do you log a map? seems like an overkill. maybe i miss the motivation

Reply:
I am using this, but now that I see it it's wrong I have to past just the "state", value:
```golang
    // Info logs a non-error message with the given key/value pairs as context. 
    //                                                                          
    // The msg argument should be used to add some constant description to      
    // the log line.  The key/value pairs can then be used to add additional    
    // variable information.  The key/value pairs should alternate string       
    // keys and arbitrary values.                                               
    Info(msg string, keysAndValues ...interface{})                              
```